### DISCUSSION TITLE
Getting while using USB Webcam alongside with Internal webcam

### DISCUSSION THREAD

**Discussion by iamjv8** [jeeliz/jeelizFaceFilter#226] [Q&A] [answered]
I am using USB webcam and my macbook's internal webcam. When i am trying to run it, It is taking USB webcam as default camera. But when I give deviceID of internal webcam in **init** method then  also it is taking USB Webcam. I have one dropdown menu to select camera to get stream but it is not switching camera with JEEFACEFILTERAPI.update_videoSettings({ 'deviceId': DEVICE_ID }) using proper deviceId. Can anyone help me in this ?

**Comment by xavierjs** [comment]
It may work better since the latest commit ( https://github.com/jeeliz/jeelizFaceFilter/commit/937f463ec3cf68bdc9c175ea6756b19f0a6e6d20 ). You also need to replace JEEFACEFILTERAPI by JEELIZFACEFILTER.

**Reply by iamjv8** [reply]
@xavierjs Thanks. 
I think this resolved my issue but it is showing other error in script file as bellow

`Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at jeelizFaceFilter.js:30
    at XMLHttpRequest.e.onreadystatechange (jeelizFaceFilter.js:38)`

**Reply by xavierjs** [reply]
Your JSON is not a JSON (see https://github.com/jeeliz/jeelizFaceFilter/issues/139)

**Reply by iamjv8** [reply]
This seems different issue... It shows me error in **jeelizFaceFilter.js** file  as bellow

![Screenshot 2021-03-24 at 8 39 06 AM](https://user-images.githubusercontent.com/53992824/112248708-67308b00-8c7c-11eb-9319-1eef46b5ab26.png)

**Reply by xavierjs** [reply]
Check your JSON using network tools, it is the basics of web dev. You are not using a HTTP server. Read the doc.

**Reply by iamjv8** [reply]
After updating my issue is resolved but I am getting 180deg rotated video stream with default settings as bellow:   
![Screenshot 2021-03-26 at 9 30 24 AM](https://user-images.githubusercontent.com/53992824/112575242-f45b1780-8e15-11eb-9dee-e3286d4bc8c8.png)

**Reply by xavierjs** [reply]
duplicate of https://github.com/jeeliz/jeelizFaceFilter/issues/213

**Reply by iamjv8** [reply]
It worked very well now. Thanks you so much for all responses @xavierjs.

**Comment by visisemanaj** [comment, marked as answer]
JEELIZFACEFILTER.init, and JEELIZFACEFILTER.update_videoSettings({ deviceId: deviceID}) don't work well for me too.
I've noticed the commit saying  **[FIX] ( Potential ) update_videoSettings() with deviceId issue** so I just pulled the latest dist/jeelizFaceFilter.module.js, but video doesn't change at all even if I change the deviceID.

**Reply by xavierjs** [reply]
Send a demo showing the bug. If I cannot reproduce I cannot help.

**Reply by visisemanaj** [reply]
Thanks, it works well today.
Do we have facefilter npm package updated too?

**Reply by xavierjs** [reply]
Hi,
I have just updated the NPM version (set to 2.1.3).

**Reply by visisemanaj** [reply]
I've noticed that when I use update_videoSettings with deviceID, it works correctly.
But if I just try to set deviceId in init without update_videoSettings, it doesn't work. It just uses default camara.

**Reply by xavierjs** [reply]
Can you submit the code chunk when you init the lib with a specific deviceId value?

**Reply by visisemanaj** [reply]
function init_faceFilter(videoSettings, canvasID, deviceID) {
  // Here we set a different pivotOffset value so that the mask fits better
  JeelizThreeHelper.set_pivotOffsetYZ(SETTINGS.pivotOffsetYZ);

  JEEFACEFILTERAPI.init({
    deviceId: deviceID, //IF deviceID is null, it uses default camera
    canvasId: canvasID,
    NNC: NN_DEFAULT,
    videoSettings: videoSettings,
    callbackReady: function (errCode, spec) {
      if (errCode) {
        console.log('AN ERROR HAPPENED. SORRY BRO :( . ERR =', errCode);
        return;
      }
      console.log('INFO: JEEFACEFILTERAPI IS READY', deviceID);
      init_threeScene(spec);

      // JEEFACEFILTERAPI.update_videoSettings({
      //   deviceId: deviceID,
      // });

      // let canvasElt = document.getElementById(canvasID);
      // canvasElt.style.visibility = 'hidden';
      //   let MediaStream = canvasElt.captureStream(25);
      //   document.getElementById('videoTag').srcObject = MediaStream;
    },

    // called at each render iteration (drawing loop)
    callbackTrack: function (detectState) {
      JeelizThreeHelper.render(detectState, THREECAMERA);
    },
  }); // end JEEFACEFILTERAPI.init call
}

Here, I've used "e3d00b04401032a24a7e345c39610e2b5f77905fbb4b2c95fec96bcff7d21613" for deviceID

**Reply by xavierjs** [reply]
You need to call:

```
JEEFACEFILTERAPI.init({
videoSettings: {deviceId: deviceID}
```

and NOT:

```
JEEFACEFILTERAPI.init({
deviceId: deviceID
```

**Reply by visisemanaj** [reply]
I see, Thanks
